### ISSUE TITLE
NoSuchMethodError on java.lang.Enum.getEnumConstants

### ISSUE THREAD

**Issue by nicolasstucki** [lampepfl/dotty#1387]
```scala
dotty> classOf[java.nio.file.AccessMode].getEnumConstants
java.lang.NoSuchMethodError: java.lang.Class.getEnumConstants()Ljava/lang/Object;
	at .<init>(<console>:8)
	at .<clinit>(<console>)
	at RequestResult$.<init>(<console>:3)
	at RequestResult$.<clinit>(<console>)
	at RequestResult$result(<console>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java...
```

This happens with any java `enum` class.

**Comment by DarkDimius** [comment]
is it indeed a `java.lang.Class.getEnumConstants()Ljava/lang/Object;`? for some reason `[` is missing.

**Comment by DarkDimius** [comment]
``` scala
        0: ldc           #15                 // class java/nio/file/AccessMode
         2: invokevirtual #21                 // Method java/lang/Class.getEnumConstants:()Ljava/lang/Object;
         5: checkcast     #23                 // class "[Ljava/nio/file/AccessMode;"
         8: areturn
```
```
package <empty> {
  @dotty.annotation.internal.SourceFile("For.scala") class FooR() extends Object
    (
  ) {
    def foo: Array[java.nio.file.AccessMode] =
      classOf[class AccessMode].getEnumConstants()
  }
}
result of For.scala after erasure:
package <empty> {
  @dotty.annotation.internal.SourceFile("For.scala") class FooR() extends Object
    (
  ) {
    def foo(): java.nio.file.AccessMode[] =
      classOf[class AccessMode].getEnumConstants().asInstanceOf[
        java.nio.file.AccessMode[]
      ]
  }
}
```

Seems like the problem is caused by incorrect erasure of `getEnumConstants`.

**Comment by smarter** [comment]
Our handling of Java `T[]` is known to be broken: https://github.com/lampepfl/dotty/issues/533

**Comment by smarter** [comment]
The culprit is indeed `eraseArray` which doesn't know that generic arrays coming from Java should be erased to `Object[]` instead of `Object` like Scala

**Comment by odersky** [comment]
Fixed by #1391.
